### DISCUSSION TITLE
Pre-accept cookiebot

### DISCUSSION THREAD

**Discussion by thany** [GoogleChrome/lighthouse#16009] [Q&A]
How can I add the right cookies to pre-accept a cookiebot popup? I don't really want to test that part.

I've already tried:

```
lighthouse <URL> --extra-headers "{\"Cookie\":\"{stamp:%27sCDdiXcadd89BFJzGGPWJYoGscKBxDDOD7bsCR8tUnuzGuDRHOFZVw==%27%2Cnecessary:true%2Cpreferences:false%2Cstatistics:false%2Cmarketing:false%2Cmethod:%27explicit%27%2Cver:3%2Cutc:1716302759779%2Cregion:%27nl%27}\"}"
```

And that is identical to a real browser having accepted it, where the popup no longer shows. But it's not working. It's either ignored, or it's wrong. It's difficult to test, as within 10 seconds or so the lighthouse browser disappears again. It's very possible that some character gets interpreted differently, like the `%` ones in there. But I have no way of knowing.

On a sidenote: why does it have to be json? That doesn't make any sense on the CLI, what with the (double) encoding going on.
(edit: *does* it have to json, actually? `--help` doesn't say what kind of format is expected, and so this is just something I found buried in examples floating around)

So anyway, what is the correct way to achieve what I want? Perhaps forcing that cookie isn't even it.

**Comment by connorjclark** [comment]
You can use a file instead:

```
lighthouse <url> --extra-headers=./path/to/file.json                                               Path to JSON file of HTTP Header key/value pairs to send in requests
```

This will let you ignore any escaping needed for your terminal, and just store the values in a plain json object.

Other approaches, FYI:

https://github.com/GoogleChrome/lighthouse/blob/main/docs/authenticated-pages.md
https://github.com/GoogleChrome/lighthouse/blob/main/docs/user-flows.md
